adapter/operations.go

```diff
@@ -83,3 +90,8 @@ func (h *Adapter) ListOperations() (Operations, error) {
 func (h *Adapter) ApplyOperation(context.Context, OperationRequest) error {
 	return nil
 }
+
+// ProcessOAM processes OAM components. This is adapter specific and needs to be implemented by each adapter.
+func (h *Adapter) ProcessOAM(context.Context, OAMRequest) error {
```

Comment:
Is there any other choice we can use,  does `panic` can work well?

Reply:
@Aisuko yeah we can panic here as we have a panic recovery mechanism. I was leaning more towards silently failing though. It would allow us to implement this function gradually/slowly in each adapter (right now only istio-adapter implements this) without returning a panic error to the client.